docker/girder.Dockerfile

```diff
@@ -9,32 +9,19 @@ COPY client/ /app/
 RUN yarn build
 
 
-FROM python:3.7-slim as runtime
-EXPOSE 8080
-# Set environment to support Unicode: http://click.pocoo.org/5/python3/#python-3-surrogate-handling
-ENV LC_ALL=C.UTF-8
-ENV LANG=C.UTF-8
-WORKDIR /girder
-COPY --from=girder/girder:latest /girder/ /girder/
-# avoid psutil GCC dependency by using unofficial large_image wheel
-RUN apt-get update \
-  && apt-get install -qy git \
-  && pip install \
-    --no-cache-dir \
-    --find-links https://girder.github.io/large_image_wheels \
-    --upgrade-strategy eager . \
-  && apt-get remove --purge -qy git \
-  && apt-get clean \
-  && rm -rf /var/lib/apt/lists/* \
-  && rm -rf /girder/.git
-COPY --from=girder/girder:latest /usr/share/girder/static/ /usr/local/share/girder/static/
+# This base image comes from an un-merged PR from Girder
+# https://github.com/girder/girder/pull/3244
+# This will become girder/girder:slim when that gets merged
+FROM gitlab.kitware.com:4567/opengeoscience/viameweb/girder:slim as runtime
 
-WORKDIR /home
-# modify this based on where you are running docker-compose from
-COPY docker/provision provision
-COPY server viame_girder
-RUN cd viame_girder && pip install --no-cache-dir .
-# Bring in the client from girder
 COPY --from=builder /app/dist/ /usr/local/share/girder/static/viame/
 
-ENTRYPOINT ["/home/provision/girder_entrypoint.sh"]
+WORKDIR /home/viame_girder
+
+COPY docker/provision /home/provision
+COPY server/setup.py /home/viame_girder/
+RUN pip install --no-cache-dir .
+COPY server/ /home/viame_girder/
+RUN pip install --no-deps .
```

Comment:
This is considered a docker python best practice, and it's the same reason you copy `package.json` and `yarn.lock` and `yarn install` before you bring the rest of the code over.

Makes use of docker caching so that you aren't reinstalling PIP dependencies when all you changed was local code.